src/Core/Checkout/Promotion/Gateway/PromotionGateway.php

```diff
@@ -34,9 +36,6 @@ public function get(Criteria $criteria, SalesChannelContext $context): EntityCol
             new FieldSorting('priority', FieldSorting::DESCENDING)
         );
 
-        /** @var PromotionCollection $entities */
-        $entities = $this->promotionRepository->search($criteria, $context->getContext())->getEntities();
-
-        return $entities;
+        return $this->promotionRepository->search($criteria, $context->getContext())->getEntities();
```

Comment:
couldn't we change the return annotation to `PromotionCollection` :thinking: 

and at the same deprecating it, so it returns `PromotionCollection` with 6.7

Reply:
Yes, I was also thinking of this. But I was not sure regarding backwards compatibility. But when thinking about it in more detail, it should be fine, if we change the return type of the implementation (`PromotionGateway`) to `PromotionCollection` and add a deprecation notice on the interface (`PromotionGatewayInterface`), that the return type will be changed to `PromotionCollection`? :thinking: 

Would that be fine? Or do we need an actual implementation where the interface is changed to return a `PromotionCollection`?

Reply:
your idea sounds good :+1: